common/auth.go

```diff
@@ -15,6 +15,41 @@
 
 package common
 
-type AuthBackend interface {
-	
-}
\ No newline at end of file
+type Role interface {
+	Name() string
+}
+
+type role struct {
```

Comment:
I know Go wants short names, but I find the presence of "Role" and "role" to be very confusing. Can we give better names to at least one of those?